packages/gatsby-plugin-clerk/src/GatsbyClerkProvider.tsx

```diff
@@ -0,0 +1,49 @@
+import { ClerkLoaded, ClerkProvider as ReactClerkProvider } from '@clerk/clerk-react';
+import { IsomorphicClerkOptions } from '@clerk/clerk-react/dist/types';
+import { navigate } from 'gatsby';
+import React from 'react';
+
+import { assertFrontendApi } from './utils';
+
+export type GatsbyClerkProviderProps = {
+  children: React.ReactNode;
+  clerkState: any;
+  frontendApi: string;
+} & IsomorphicClerkOptions;
+
+export function ClerkProvider({ children, ...rest }: GatsbyClerkProviderProps) {
+  const { clerkJSUrl, clerkState, frontendApi, ...restProps } = rest;
+  const { __clerk_ssr_state, __clerk_ssr_interstitial } = clerkState?.__internal_clerk_state || {};
+
+  assertFrontendApi(frontendApi);
+
+  return (
+    <ReactClerkProvider
+      frontendApi={frontendApi}
+      clerkJSUrl={clerkJSUrl}
+      navigate={to => navigate(to)}
+      initialState={__clerk_ssr_state || {}}
+      {...restProps}
+    >
+      {__clerk_ssr_interstitial ? (
+        <ClerkLoaded>
+          <Reload />
+        </ClerkLoaded>
+      ) : (
+        children
+      )}
+    </ReactClerkProvider>
+  );
+}
+
+const Reload = () => {
+  React.useEffect(() => {
+    // TODO: Add fix for firefox edge case
```

Comment:
❓ 
The way to fix this is already in the latest interstitial code, right, can't we use the same ?

Reply:
Thanks for the review @igneel64 ! Yes you're right, its going to be added immediately after the beta is released. The next version will include the FF fix and observability